### PR TITLE
Fixes alignment of the select;

### LINKED ISSUE #194: Select: component does not align properly inline
When inline text is added next to the select component, it does not line up properly. Specifically, for use cases where the label might be needed.

image

### PR DESCRIPTION
**Pull request by seangates**
## Description
This fixes some styles related to the inline view of the select component. <strike>It also adds a `--small` modifier for when the select needs to match the default font size.</strike>

## Context
When placed inline with text, spans, or other inline content the select would position itself incorrectly.

## References
Fixes #194 

### Before
image

### After
image

### CODE REVIEW

**Comment by ianmcburnie** [comment]
@seangates The caret seems a bit too low to me:

screen shot 2018-05-09 at 10 53 35 am

Seems to be caused by the margin-top: -1px?

Without the margin-top: -1px:

screen shot 2018-05-09 at 10 55 24 am

Which is correct according to DSL?

**Comment by seangates** [comment]
@ianmcburnie I guess there are competing concerns. The margin change is for the text line heights to line up, whereas the caret is being aligned to the box (not the text). Let me look at things.

File: src/less/select/ds6/select-base.less [hunk 1]
```diff
@@ -22,8 +23,10 @@
     background-color: #fff;
     border: 1px solid #c7c7c7;
     border-radius: 0;
+    font-family: @ds6-font-family-custom; // for some reason this needs to be explicitly set
```
[the remark this answers is not in the document]
Comment (seangates) [reply src/less/select/ds6/select-base.less]: Nope.

**Comment by ianmcburnie** [comment]
Going to leave this for 5.1.0

File: dist/select/ds6/select.css [hunk 2]
```diff
@@ -26,8 +27,10 @@
   background-color: #fff;
   border: 1px solid #c7c7c7;
   border-radius: 0;
+  font-family: "Market Sans", Arial, sans-serif;
```
Comment (ianmcburnie) [inline dist/select/ds6/select.css]: @seangates @RajaRamu @senthilp  Will this interfere with how the ebay-font loader works? Will setting Market Sans explicitly in our CSS cause issues? I know we have avoided setting it explicitly in our global module, so just want to make sure here.

I also have another case where we need to set it explicitly for `button.fake-menu__item`.

Comment (yomed) [reply dist/select/ds6/select.css]: Did you try `inherit` for the button case? I briefly tried it when I filed that issue, and it seemed to work.

Comment (senthilp) [reply dist/select/ds6/select.css]: @ianmcburnie As @yomed mentioned we should first try with `inherit`. Setting Market Sans explicitly will cause FOUT/FOIT in some browsers, as `font-display` [support](https://caniuse.com/#search=font-display) is still not widely available.

Comment (senthilp) [reply dist/select/ds6/select.css]: Looks like resolved now 👍

Comment (ianmcburnie) [reply dist/select/ds6/select.css]: It's resolved for the other issue in menu, but not here for select.

Comment (seangates) [reply dist/select/ds6/select.css]: Correct. For one reason or another, the `<select>` doesn't inherit it from anywhere and it needs to be explicitly set.

Comment (senthilp) [reply dist/select/ds6/select.css]: The other solution would be to use a more specific selector, just for `font-family`
```css
.font-marketsans .select > select {
    font-family: "Market Sans", Arial, sans-serif;
}
```
[`ebay-font`](https://github.com/senthilp/ebay-font/blob/master/font/marketsans/template.marko#L11) adds that class `font-marketsans` only when Market Sans is available and it should get the best of both worlds.

Comment (seangates) [reply dist/select/ds6/select.css]: @senthilp I was not aware of the `.font-marketsans` class. I will go ahead and add that selector.

**Comment by seangates** [comment]
Why was this closed?

**Comment by ianmcburnie** [comment]
Accident! Sorry!

**Comment by PrashantAshok** [comment]
@ianmcburnie - did you get a chance to look why `/docs/static/ds6/docs.js` shows up changes with 5.1.0 branch?

**Comment by seangates** [comment]
@ianmcburnie @PrashantAshok This should be ready.

**Comment by ianmcburnie** [comment]
@seangates Just been looking at the ds4 select (aka listbox) and notice we don't have vertical-align there, yet it works okay (even without field class). What's the difference with ds6, why do we need vertical-align now? What am I missing?

**Comment by seangates** [comment]
DS4 _does_ still have the issue, just not as pronounced.

### DS4 with no `vertical-align: middle;`
image

### DS4 _with_ `vertical-align: middle;`
image

**Comment by ianmcburnie** [comment]
Oh, my bad. I was talking about the alignment of the label. Given the original screenshot, I thought you had added vertical align specifically for that...?

**Comment by ianmcburnie** [comment]
Do you still see that non-label related issue on the test page:

http://ebay.github.io/skin/test/misc/ds4/alignment-controls-inline.html

**Comment by seangates** [comment]
👍 Glad we're on the same page now.

**Comment by ianmcburnie** [comment]
Well, I wouldn't go that far! Still needs some clearing up...

So, answering my own question, it seems that `.select > select { vertical-align: top }` is the difference between ds4 and ds6 causing inline *text* to be aligned incorrectly.

screen shot 2018-06-05 at 4 53 25 pm

Removing `vertical-align: top`:

screen shot 2018-06-05 at 4 53 33 pm

Now adding a button:

screen shot 2018-06-05 at 4 54 23 pm

Now removing `vertical-align: middle` from button:

screen shot 2018-06-05 at 4 55 05 pm

Looks good to me.

Keeping `vertical-align: middle` on .btn and adding `vertical-align: middle` to select:

screen shot 2018-06-05 at 4 57 02 pm

Doesn't look as good to me (the label is higher).

So with that quick experiment it would seem dropping vertical-align: top from the select element and vertical-align: middle from the .btn class might be the way to go. I only tested in chrome and safari though.

**Comment by ianmcburnie** [comment]
I say for now we just remove `vertical-align: top` and then we can tackle inline alignment of controls in a future PR.

**Comment by ianmcburnie** [comment]
And we could put this into 5.0.2 patch if you like...?

**Comment by seangates** [comment]
So, removing the `vertical-align: top;` yields this:
image

Blech.

**Comment by ianmcburnie** [comment]
Yeah, I showed that in my comment above. Removing `vertical-align: middle` from the button will line these items up. But for this PR let's just remove the `vertical align: top` which addresses the issue in the original comment screenshot.

I don't want to modify the `.btn` class just yet until I've done a full sweep of other controls, across browsers. For example, I seem to remember that an anchor tag button (i.e, a fake button) did need vertical align middle.

**Comment by seangates** [comment]
Closing to branch from 5.0.2.

**Pull request closed without merging**